File: solution/2134. Minimum Swaps to Group All 1's Together II/2134.cpp
class Solution {
public:
    int minSwaps(vector<int>& nums) {
        int ones = count(nums.begin(), nums.end(), 1), n = nums.size(), res = nums.size();
        for (int i = 0, j = 0, cnt = 0; i < n; ++i) {
            while (j - i < ones)
                cnt += nums[j++ % n];
            res = min(res, ones - cnt);
            cnt -= nums[i];
        }
        return res;
    }
};


// x x [1 0 1] x x
// [x x x] x x x x

File: solution/2134. Minimum Swaps to Group All 1's Together II/2134.py
class Solution:
    def minSwaps(self, nums: List[int]) -> int:
        ones = nums.count(1)
        n = res = len(nums)
        j = cnt = 0
        for i in range(n):
            while j - i < ones:
                cnt += nums[j % n]
                j += 1
            res = min(res, ones - cnt)
            cnt -= nums[i]
        return res
    
